src/main/scala/csvreader/CSVReader.scala

```diff
@@ -1,28 +1,43 @@
 package csvreader
 
-import scala.collection.mutable.Map
+import scala.collection.mutable
 import scala.concurrent.Future
 import scala.concurrent.ExecutionContext.Implicits.global
-import scala.util.{Failure, Success}
+import scala.io.BufferedSource
+import scala.util.{Try, Failure, Success}
 
 /** CSVReader
   * 
   * 
   */
 object CSVReader extends App {
-    val file = io.Source.fromFile("people.csv") 
-    var m = Map[String, Int]()
+  var sourcePath = "people.csv"
+  if (args.length > 0)
+    sourcePath = args(0)
+  val readFile = Try(io.Source.fromFile(sourcePath))
+  var m = mutable.Map[String, Int]()
+  readFile match {
+    case Success(file) => m = countStates(file)
+    case Failure(exception) => println("Please pass a valid source path with sbt \"run [sourcePath]\".")
+  }
+  
+  m.foreach(println)
+
+  val future = Future {
+    m.foreach(println)
+  }
+
+  /** Returns a map of states with their associated counts. */
+  def countStates(file: BufferedSource): mutable.Map[String, Int] = {
+    var m = mutable.Map[String, Int]()
     for (line <- file.getLines) {
       var state = line.split(",")(2)
-      if (m.contains(state)) {
+      if (m.contains(state))
         m(state) += 1
-      } else {
```

Comment:
Removing '{' might not be best practice even though it works

Reply:
According to this, it's best practice when you have an 'else' clause: https://docs.scala-lang.org/style/control-structures.html#curly-braces